tests/test_csc.py

```diff
@@ -815,6 +817,91 @@ async def test_show_alarms(self):
+
+            await salobj.set_summary_state(
+                self.remote, state=salobj.State.ENABLED, override="enabled.yaml"
+            )
+            await self.check_all_alarms_events_are_none()
+
+            await atdome.evt_summaryState.set_write(
+                summaryState=salobj.State.FAULT, force_output=True
+            )
+            await self.assert_next_alarm(
+                name=atdome_alarm_name,
+                severity=AlarmSeverity.CRITICAL,
+                maxSeverity=AlarmSeverity.CRITICAL,
+                acknowledged=False,
+                acknowledgedBy="",
+            )
+
+            test_case_json = {"key": "TEST-123", "status": "PASS"}
+            mock_response = AsyncMock()
+            mock_response.json.return_value = test_case_json
+            mock_post.return_value.__aenter__.return_value = mock_response
+
+            test_response = await self.remote.cmd_makeLogEntry.set_start(
+                name=atdome_alarm_name,
+                timeout=STD_TIMEOUT,
+            )
+
+            # The actual test_response will have more fields than expected.
+            # We don't care about most of them and they might cause variations
+            # in results so construct a subset of the the actual for only the
+            # keys in expected.
+            expected_response = {
+                "ack": 303,
+                "error": 0,
+                "result": "Done",
+                "identity": "Watcher",
+                "cmdtype": 5,
+            }
+            actual_subset = dict_subset(
+                expected_response, test_response.__dict__.items()
+            )
+            self.assertEqual(actual_subset, expected_response)
+            post_args = mock_post.call_args.kwargs
+            assert post_args["url"] == "https://fake.url.com/narrativelog/messages"
+            assert (
+                post_args["json"]["message_text"]
+                == "alarm:Enabled.ATDome:0 severity=4 FAULT state"
+            )
+            expected = {
+                "url": "https://fake.url.com/narrativelog/messages",
+                "json": {
+                    "message_text": "alarm:Enabled.ATDome:0 severity=4 FAULT state",
+                    "level": 50,
+                    "user_id": "saluser@chimp20",
+                    "user_agent": "Watcher",
+                    "is_human": False,
+                    "tags": ["watcher", "alarm", "make_log_entry"],
+                    "date_begin": "2024-08-22T15:22:37.444279",
+                    "date_end": "2024-08-22T15:22:37.444279",
+                },
+            }
+            self.assertIn("user_id", post_args["json"])
+            self.assertIn("date_begin", post_args["json"])
+            self.assertIn("date_end", post_args["json"])
+            del post_args["json"]["user_id"]
+            del post_args["json"]["date_begin"]
+            del post_args["json"]["date_end"]
+            del expected["json"]["user_id"]
```

Comment:
Instead of adding and then deleting, just don't add these to the expected dict.

Reply:
I would prefer to have the full data  that is returned in expected. It IS what is expected, after all.  Its just that we are less picky about a few of the expected values.  Think keeping Expected and Actual (post_args) in sync makes things more obvious and more maintainable.

Reply:
Well, but the problem is you "expected" is not really what you expected right? The dates are dynamic so your expectation is that the fields will exists, but you can't determine the values.

Think about someone looking at this in the future. They will wonder, why there's this hard-coded date here in the "expected" dictionary and why it is being deleted afterwards?

Reply:
Yes, they might wonder.  I would actually prefer to comment about the part of the structure in expected that we are not using.  But I thought you would object to such commented code.  Personally, if I saw a hard-coded date, I would think:  "Oh, of course, it varies so cannot compare to it.  But I'm glad they showed me an example of it so I know exactly the format."

Reply:
Commenting code is against the guidelines so, yeah, probably not a good idea. However, you could make a comment just above the `expected` dictionary saying that the dictionary also contains the additional keys that are dynamic, and therefore, removed from the expected dictionary.